llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -2136,6 +2136,18 @@ bool AArch64LoadStoreOpt::tryToPairLdStInst(MachineBasicBlock::iterator &MBBI) {
   if (!TII->isCandidateToMergeOrPair(MI))
     return false;
 
+  // Fetch the memoperand of the load/store that is a candidate for
+  // combination.
+  MachineMemOperand *MemOp = MI.memoperands().front();
```

Comment:
This may have to do with the regression on x64, because its a memoperand assertion. Although I am still confused a bit. Thanks for the hint! Should help a lot!